Hatch for mypy still not totally reproducible?

### Describe the bug

I've always had unit test errors and lint failures that do not pop up on CI.
I chalked it up to maybe some discrepancies in env

But now with hatch, I'm still having the same problems?

Is it possible to use dagger locally?

### Environment

`hatch run mypy marimo/`

<details>
```text
marimo/_data/charts.py:80: error: Module has no attribute "datum"  [attr-defined]
marimo/_tracer.py:17: error: Cannot find implementation or library stub for module named "opentelemetry"  [import-not-found]
marimo/_tracer.py:90: error: Cannot find implementation or library stub for module named "opentelemetry.sdk.trace"  [import-not-found]
marimo/_tracer.py:91: error: Cannot find implementation or library stub for module named "opentelemetry.sdk.trace.export"  [import-not-found]
marimo/_tracer.py:103: error: Class cannot subclass "SpanExporter" (has type "Any")  [misc]
marimo/_output/formatters/altair_formatters.py:72: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_output/formatters/altair_formatters.py:80: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_output/formatters/altair_formatters.py:81: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_output/formatters/altair_formatters.py:104: error: Module has no attribute "renderers"  [attr-defined]
marimo/_output/formatters/altair_formatters.py:105: error: Module has no attribute "Undefined"  [attr-defined]
marimo/_runtime/runner/hooks_preparation.py:14: error: Untyped decorator makes function "_update_stale_statuses" untyped  [misc]
marimo/_runtime/runner/hooks_pre_execution.py:13: error: Untyped decorator makes function "_set_staleness" untyped  [misc]
marimo/_runtime/runner/hooks_pre_execution.py:30: error: Untyped decorator makes function "_set_status_to_running" untyped  [misc]
marimo/_runtime/app/kernel_runner.py:125: error: Returning Any from function declared to return "bool"  [no-any-return]
marimo/_runtime/app/kernel_runner.py:131: error: Returning Any from function declared to return "tuple[HumanReadableStatus, JSONType, bool]"  [no-any-return]
marimo/_plugins/ui/_impl/tables/pyarrow_table.py:65: error: "Array" expects 2 type arguments, but 1 given  [type-arg]
marimo/_plugins/ui/_impl/tables/pyarrow_table.py:65: error: Missing type parameters for generic type "Array"  [type-arg]
marimo/_plugins/ui/_impl/tables/pyarrow_table.py:67: error: "Array" expects 2 type arguments, but 1 given  [type-arg]
marimo/_plugins/ui/_impl/tables/pyarrow_table.py:67: error: Missing type parameters for generic type "Array"  [type-arg]
marimo/_plugins/ui/_impl/tables/pyarrow_table.py:108: error: Argument 1 to "record_batch" has incompatible type "list[Array[Any, Any]]"; expected "list[Array[Any, Any] | ChunkedArray[Any, Any]]"  [arg-type]
marimo/_plugins/ui/_impl/tables/pyarrow_table.py:108: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
marimo/_plugins/ui/_impl/tables/pyarrow_table.py:108: note: Consider using "Sequence" instead, which is covariant
marimo/_plugins/ui/_impl/tables/pyarrow_table.py:276: error: "ChunkedArray" expects 2 type arguments, but 1 given  [type-arg]
marimo/_plugins/ui/_impl/tables/pyarrow_table.py:276: error: Missing type parameters for generic type "ChunkedArray"  [type-arg]
marimo/_plugins/ui/_impl/charts/altair_transformer.py:90: error: Module has no attribute "sanitize_pandas_dataframe"; maybe "sanitize_dataframe"?  [attr-defined]
marimo/_plugins/ui/_impl/charts/altair_transformer.py:141: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_plugins/ui/_impl/charts/altair_transformer.py:142: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_plugins/ui/_impl/charts/altair_transformer.py:143: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_plugins/ui/_impl/charts/altair_transformer.py:144: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_runtime/runner/hooks_post_execution.py:42: error: Untyped decorator makes function "_set_imported_defs" untyped  [misc]
marimo/_runtime/runner/hooks_post_execution.py:58: error: Untyped decorator makes function "_set_status_idle" untyped  [misc]
marimo/_runtime/runner/hooks_post_execution.py:69: error: Untyped decorator makes function "_set_run_result_status" untyped  [misc]
marimo/_runtime/runner/hooks_post_execution.py:85: error: Untyped decorator makes function "_broadcast_variables" untyped  [misc]
marimo/_runtime/runner/hooks_post_execution.py:105: error: Untyped decorator makes function "_broadcast_datasets" untyped  [misc]
marimo/_runtime/runner/hooks_post_execution.py:124: error: Untyped decorator makes function "_broadcast_duckdb_tables" untyped  [misc]
marimo/_runtime/runner/hooks_post_execution.py:155: error: Untyped decorator makes function "_store_reference_to_output" untyped  [misc]
marimo/_runtime/runner/hooks_post_execution.py:189: error: Untyped decorator makes function "_broadcast_outputs" untyped  [misc]
marimo/_runtime/runner/hooks_post_execution.py:291: error: Untyped decorator makes function "_reset_matplotlib_context" untyped  [misc]
marimo/_runtime/runner/hooks_on_finish.py:25: error: Untyped decorator makes function "_send_interrupt_errors" untyped  [misc]
marimo/_runtime/runner/hooks_on_finish.py:42: error: Untyped decorator makes function "_send_cancellation_errors" untyped  [misc]
marimo/_plugins/ui/_impl/altair_chart.py:271: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_plugins/ui/_impl/altair_chart.py:277: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_plugins/ui/_impl/altair_chart.py:280: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_plugins/ui/_impl/altair_chart.py:408: error: Argument 1 to "_has_geoshape" has incompatible type "dict[str, Any]"; expected "TopLevelMixin"  [arg-type]
marimo/_plugins/ui/_impl/altair_chart.py:440: error: Name "altair.UndefinedType" is not defined  [name-defined]
marimo/_plugins/ui/_impl/altair_chart.py:472: error: Module "altair" has no attribute "Undefined"  [attr-defined]
marimo/_plugins/ui/_impl/altair_chart.py:501: error: Incompatible types in assignment (expression has type "DataFrameLike | None", variable has type "DataFrame")  [assignment]
marimo/_plugins/ui/_impl/altair_chart.py:587: error: Module "altair" has no attribute "Undefined"  [attr-defined]
marimo/_plugins/ui/_impl/altair_chart.py:611: error: Module has no attribute "Undefined"  [attr-defined]
marimo/_data/preview_column.py:122: error: Module "altair" has no attribute "MaxRowsError"  [attr-defined]
marimo/_data/preview_column.py:122: note: Error code "attr-defined" not covered by "type: ignore" comment
marimo/_data/preview_column.py:153: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_data/preview_column.py:159: error: Module has no attribute "data_transformers"  [attr-defined]
marimo/_runtime/runtime.py:536: error: Untyped decorator makes function "code_completion" untyped  [misc]
marimo/_runtime/runtime.py:1238: error: Untyped decorator makes function "delete_cell" untyped  [misc]
marimo/_runtime/runtime.py:1249: error: Untyped decorator makes function "run" untyped  [misc]
marimo/_runtime/runtime.py:1292: error: Untyped decorator makes function "rename_file" untyped  [misc]
marimo/_runtime/runtime.py:1302: error: Untyped decorator makes function "run_scratchpad" untyped  [misc]
marimo/_runtime/runtime.py:1343: error: Untyped decorator makes function "run_stale_cells" untyped  [misc]
marimo/_runtime/runtime.py:1359: error: Untyped decorator makes function "set_cell_config" untyped  [misc]
marimo/_runtime/runtime.py:1384: error: Untyped decorator makes function "set_user_config" untyped  [misc]
marimo/_runtime/runtime.py:1388: error: Untyped decorator makes function "set_ui_element_value" untyped  [misc]
marimo/_runtime/runtime.py:1582: error: Untyped decorator makes function "function_call_request" untyped  [misc]
marimo/_runtime/runtime.py:1677: error: Untyped decorator makes function "instantiate" untyped  [misc]
marimo/_runtime/runtime.py:1788: error: Untyped decorator makes function "preview_dataset_column" untyped  [misc]
marimo/_server/sessions.py:921: error: Untyped decorator makes function "start" untyped  [misc]
marimo/_cli/upgrade.py:35: error: Untyped decorator makes function "check_for_updates" untyped  [misc]
marimo/_server/api/middleware.py:116: error: Cannot find implementation or library stub for module named "opentelemetry"  [import-not-found]
marimo/_server/api/middleware.py:117: error: Cannot find implementation or library stub for module named "opentelemetry.trace.status"  [import-not-found]
marimo/_server/api/middleware.py:117: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 66 errors in 16 files (checked 326 source files)
```
</details>

### Code to reproduce

_No response_

--- a/scripts/pycheck.sh
+++ b/scripts/pycheck.sh
@@ -9,4 +9,4 @@ hatch run ruff check --fix
 echo "[check: format]"
 hatch run ruff format
 echo "[check: typecheck]"
-hatch run mypy marimo/
+hatch run typecheck:check

